src/ArenaReplay.cpp

```diff
@@ -252,33 +259,144 @@ class ArenaReplayBGScript : public BGScript
+            ArenaTeam* team = sArenaTeamMgr->GetArenaTeamById(bg->GetArenaTeamIdForTeam(bgTeamId));
+            uint32 arenaTeamId = bg->GetArenaTeamIdForTeam(bgTeamId);
+            TeamId teamId = static_cast<TeamId>(arenaTeamId);
+            uint32 teamMMR = bg->GetArenaMatchmakerRating(teamId);
+
+            if (bgTeamId == winnerTeamId)
+            {
+                if (!winnerClassIds.empty())
+                    winnerClassIds += ", ";
+                winnerClassIds += playerClassId;
+
+                if (!winnerPlayerNames.empty())
+                    winnerPlayerNames += ", ";
+                winnerPlayerNames += playerName;
+
+                if (bg->isRated() && team)
+                {
+                    if (team->GetId() < 0xFFF00000)
+                    {
+                        teamWinnerName = team->GetName();
+                        teamWinnerRating = team->GetRating();
+                        teamWinnerMMR = teamMMR;
+                    }
+                    // 3v3 Solo Queue match (temporary team to merge players in 1 team)
+                    else if (team->GetId() >= 0xFFF00000)
+                    {
+                        teamWinnerName = "3v3 Solo Queue";
+                        teamWinnerRating = team->GetRating();
+                        teamWinnerMMR = teamMMR;
+                    }
+                }
+                if (bg->isArena() && !bg->isRated())
+                {
+                    teamWinnerName = "Skirmish Arena";
+                }
+                else if (!bg->isArena())
+                {
+                    teamWinnerName = "Battleground";
+                }
+            }
+            else // Loss
+            {
+                if (!loserClassIds.empty())
+                    loserClassIds += ", ";
+                loserClassIds += playerClassId;
+
+                if (!loserPlayerNames.empty())
+                    loserPlayerNames += ", ";
+                loserPlayerNames += playerName;
+
+                if (bg->isRated() && team)
+                {
+                    if (team->GetId() < 0xFFF00000)
+                    {
+                        teamLoserName = team->GetName();
+                        teamLoserRating = team->GetRating();
+                        teamLoserMMR = teamMMR;
+                    }
+                    // 3v3 Solo Queue match
+                    else if (team->GetId() >= 0xFFF00000)
+                    {
+                        teamLoserName = "3v3 Solo Queue";
+                        teamLoserRating = team->GetRating();
+                        teamLoserMMR = teamMMR;
+                    }
+                }
+                if (bg->isArena() && !bg->isRated())
+                {
+                    teamLoserName = "Skirmish Arena";
+                }
+                else if (!bg->isArena())
+                {
+                    teamLoserName = "Battleground";
+                }
+            }
```

Comment:
this code seems a bit redundant, you could save all the data in some `tmpTeam*` variables (loser or winner) and then later make simpler association with an if, something like this:
```cpp
std::string tmpTeamName;
uint32 tmpTeamRating = 0;
uint32 tmpTeamMMR = 0;

if (bg->isRated() && team)
{
    if (team->GetId() < 0xFFF00000)
    {
        tmpTeamName = team->GetName();
        tmpTeamRating = team->GetRating();
        tmpTeamMMR = teamMMR;
    }
    // 3v3 Solo Queue match (temporary team to merge players in 1 team)
    else if (team->GetId() >= 0xFFF00000)
    {
        tmpTeamName = "3v3 Solo Queue";
        tmpTeamRating = team->GetRating();
        tmpTeamMMR = teamMMR;
    }
}
if (bg->isArena() && !bg->isRated())
{
    tmpTeamName = "Skirmish Arena";
}
else if (!bg->isArena())
{
    tmpTeamName = "Battleground";
}

if (bgTeamId == winnerTeamId) {
    if (!winnerClassIds.empty())
        winnerClassIds += ", ";
    winnerClassIds += playerClassId;

    if (!winnerPlayerNames.empty())
        winnerPlayerNames += ", ";
    winnerPlayerNames += playerName;

  teamWinnerName = tmpTeamName;
  teamWinnerRating = tmpTeamRating;
  teamWinnerMMR = teamMMR;
} else {
    if (!loserClassIds.empty())
        loserClassIds += ", ";
    loserClassIds += playerClassId;

    if (!loserPlayerNames.empty())
        loserPlayerNames += ", ";
    loserPlayerNames += playerName;

  teamLoserName = tmpTeamName;
  teamLoserRating = tmpTeamRating;
  teamLoserMMR = teamMMR;
}
```